server/e2e_test.go

```diff
@@ -0,0 +1,279 @@
+package server
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"net"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/cashshuffle/cashshuffle/message"
+)
+
+var (
+	testTempKey int
+)
+
+func TestHappyShuffle(t *testing.T) {
+	poolSize := 3
+	h := testNewHarness(t, poolSize)
+	bch1 := uint64(100000000)
+	version := uint64(999)
+
+	// clients join the pool, one at a time
+	clients := make([]*testClient, 0)
+	for i := 0; i < poolSize; i++ {
+		client := h.NewClient()
+		clients = append(clients, client)
+
+		client.Register(bch1, version)
+		if i < poolSize-1 {
+			h.AssertBroadcastNewPlayer(client, clients)
```

Comment:
MIght want to test that the pool is filling up with the correct data, and is not full. =)

Reply:
8ced73a

Good enough? What kind of correct data are you thinking of? I am definitely considering expanding the server state scope.

Reply:
This looks great. =)